### PR TITLE
Staging to main

### PR DESCRIPTION
**Pull request by OP-ReubenK**
Staging needs to be merged to main. The previous main has been nuked as it was very out dated. This is our current code that we will be leaving for the next team, so we all need to be happy with it.

### CODE REVIEW

File: react-arcade/src/components/Game.jsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
+// import React from "react";
```
Comment (fuzzyers) [inline react-arcade/src/components/Game.jsx:1]: This file could be deleted

**Review by fuzzyers — APPROVED** [review]
I have confirmed that Both the arcade frontend and arcade backend are still working together.

File: react-arcade/src/utils/Arcade-Controls.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import Item from "../components/Game-Item";
+import { CSSTransition } from "react-transition-group"
+import { launchGame } from "./Arcade-Backend-Functions";
+
```
Comment (TomMc608) [inline react-arcade/src/utils/Arcade-Controls.js:4]: a comment on this code would be good just a quick description on what the code is being used for.

**Review by ghost — CHANGES_REQUESTED** [review]
Code looks really good. there are some bits where commenting would be good just to explain what that part of the code does. we also have lots of files that look like they were used for testing, we should delete them and any unnecessary assets.

File: react-arcade/src/components/App.scss [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+}
+
+.launchBar {
+  display: none;
+}
+
+.game-item {
+  // display: flex;
+  // justify-content: center;
+  // background-color: darkorange;
```
[the remark this answers is not in the document]
Comment (OP-ReubenK) [reply react-arcade/src/components/App.scss:105]: this is in the wiki https://github.com/OtagoPolytechnic/third-year-studio-2024-rainy-arcade/wiki/Arcade-game-select#game-item

File: controller-select/Keyboard1.gamecontroller.amgp [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+                <slots>
+                    <slot>
+                        <code>0x55</code>
+                        <mode>keyboard</mode>
+                    </slot>
+                </slots>
+            </button>
+        </set>
+    </sets>
+</gamecontroller>
```
Comment (SamanthaCra) [inline controller-select/Keyboard1.gamecontroller.amgp:162]: Same as above, comments would be good.

File: controller-select/Space Fight.txt [hunk 5]
```diff
@@ -0,0 +1,5 @@
+Dev Name: Tom
```
[the remark this answers is not in the document]
Comment (SamanthaCra) [reply controller-select/Space Fight.txt:1]: This should be deleted

File: controller-select/identify.py [hunk 6]
```diff
@@ -0,0 +1,28 @@
+import subprocess
+import re
+antimicrox_path = r"C:\Program Files\AntiMicroX\bin\antimicrox.exe"
```
Comment (SamanthaCra) [inline controller-select/identify.py:3]: We'll need to double check this is the correct path on the machine.

File: file-upload/src/App.js [hunk 7]
```diff
@@ -24,45 +54,61 @@ const Form = ({ name, setName, color, setColor, onSubmit }) => (
...
-
 function App() {
-  const [file, setFile] = useState(null);
-  const [name, setName] = useState('');
+  const [zipFile, setZipFile] = useState(null);
+  const [imageFile, setImageFile] = useState(null);
+  const [devName, setDevName] = useState('');
+  const [gameName, setGameName] = useState('');
   const [color, setColor] = useState('#000000');
+  const [controller, setController] = useState('Keyboard');
```
Comment (SamanthaCra) [inline file-upload/src/App.js:63]: Is this meant to be hard coded to keyboard?

File: react-arcade/assets/games/old/BadPac/BadPac.exe
Comment (SamanthaCra) [inline react-arcade/assets/games/old/BadPac/BadPac.exe:1]: Does this still need to be here?

File: react-arcade/assets/games/old/BadPac/Space Fight.jfif
Comment (SamanthaCra) [inline react-arcade/assets/games/old/BadPac/Space Fight.jfif:1]: Does this still need to be here?

File: react-arcade/assets/games/old/Doom/BadPac.exe
Comment (SamanthaCra) [inline react-arcade/assets/games/old/Doom/BadPac.exe:1]: Does this still need to be here?

File: react-arcade/assets/games/young/YG1/BadPac.exe
Comment (SamanthaCra) [inline react-arcade/assets/games/young/YG1/BadPac.exe:1]: Does this still need to be here?

File: react-arcade/assets/img/logo_gb_cart.png
Comment (SamanthaCra) [inline react-arcade/assets/img/logo_gb_cart.png:1]: Does this still need to be here?

File: react-arcade/src/assets/img/sacredKrow.png
Comment (SamanthaCra) [inline react-arcade/src/assets/img/sacredKrow.png:1]: Does this still need to be here? ALso named wrong

**Review by SamanthaCra — CHANGES_REQUESTED** [review]
Looking good, just needing more comments I think mainly from @fuzzyers and some of the redundant testing removed in order to clean up the branch.

File: controller-select/identify.py (hunk 6 above)
Comment (OP-ReubenK) [reply controller-select/identify.py:3]: I pulled from Staging today (12/06) and it works well

File: arcadebackend/index.js [hunk 8]
```diff
@@ -1,39 +1,161 @@
...
+    // Ensure the required parameters are provided
+    if (!relativePath || !antiMicroPath) {
+        res.status(400).send('Missing required parameters');
+        isRunning = false;
         return;
+    }
+
+    const newDirectory = path.join(DIRECTORY, antiMicroPath);
+
+    // process.chdir(newDirectory);
```
[the remark this answers is not in the document]
Comment (fuzzyers) [reply arcadebackend/index.js:36]: image

I have updated it with a comment explaining it for the next developer aswell as removing line by line comments

File: react-arcade/src/components/Game.jsx (hunk 1 above)
Comment (fuzzyers) [reply react-arcade/src/components/Game.jsx:1]: I have Removed this file as it was unused

File: controller-select/Space Fight.txt (hunk 5 above)
[the remark this answers is not in the document]
Comment (fuzzyers) [reply controller-select/Space Fight.txt:1]: It has been deleted

File: move/AO/Space FightTeen/Space Fight.jfif
[the remark this answers is not in the document]
Comment (fuzzyers) [reply move/AO/Space FightTeen/Space Fight.jfif:1]: It has been deleted

File: move/AO/Space FightTeen/Space Fight.txt [hunk 9]
```diff
@@ -0,0 +1,5 @@
+Dev Name: Tom
```
[the remark this answers is not in the document]
Comment (fuzzyers) [reply move/AO/Space FightTeen/Space Fight.txt:1]: It has been deleted

File: move/E/Space Fight1/Space Fight.txt [hunk 10]
```diff
@@ -0,0 +1,5 @@
+Dev Name: Tom
```
[the remark this answers is not in the document]
Comment (fuzzyers) [reply move/E/Space Fight1/Space Fight.txt:1]: It has been deleted

File: react-arcade/assets/games/old/BadPac/test1.txt
[the remark this answers is not in the document]
Comment (fuzzyers) [reply react-arcade/assets/games/old/BadPac/test1.txt:1]: I have Deleted the games folder as none of it is being used now

File: react-arcade/public/assets/games/old/Game 5/test1.txt
[the remark this answers is not in the document]
Comment (fuzzyers) [reply react-arcade/public/assets/games/old/Game 5/test1.txt:1]: I have removed the games directory

File: react-arcade/src/components/Game.jsx [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+// import "./App.scss";
+
+// const Game = () => {
```
[the remark this answers is not in the document]
Comment (fuzzyers) [reply react-arcade/src/components/Game.jsx:4]: That file was no longer being used in the code base so I have removed it

File: react-arcade/src/utils/Arcade-Controls.js (hunk 2 above)
Comment (fuzzyers) [reply react-arcade/src/utils/Arcade-Controls.js:4]: This code was actually updated and replaced and has now been removed from the code base

**Review by SamanthaCra — APPROVED** [review]
Thanks for fixing that @fuzzyers !

**Review by ramandeepkau — COMMENTED** [review]
Overall, the changes look solid and well-implemented. There are no major conflicts or issues. Great job!"

**Pull request merged**